compiler/src/dotty/tools/dotc/transform/InstrumentCoverage.scala

```diff
@@ -0,0 +1,312 @@
+              thenp = instrument(coverageTransform(tree.thenp), branch = true),
+              elsep = instrument(coverageTransform(tree.elsep), branch = true)
+            )
+          case tree: Try =>
+            cpy.Try(tree)(
+              expr = instrument(coverageTransform(tree.expr), branch = true),
+              cases = transformCases(tree.cases, forceInstrument),
+              finalizer = instrument(coverageTransform(tree.finalizer), true)
+            )
+
+          // a.f(args)
+          case tree @ Apply(fun: Select, args) =>
+            // don't transform the first Select, but do transform `a.b` in `a.b.f(args)`
+            if canInstrumentApply(tree) then
+              val transformedFun = cpy.Select(fun)(transform(fun.qualifier), fun.name)
+              if needsLift(tree) then
+                val transformed = cpy.Apply(tree)(transformedFun, args) // args will be transformed in instrumentLifted
+                instrumentLifted(transformed)
+              else
+                val transformed = cpy.Apply(tree)(transformedFun, transform(args))
+                instrument(transformed)
+            else
+              tree
+
+          // f(args)
+          case tree: Apply =>
+            if canInstrumentApply(tree) then
+              if needsLift(tree) then
+                instrumentLifted(tree)
+              else
+                instrument(super.transform(tree))
+            else
+              tree
+
+          // (f(x))[args]
+          case TypeApply(fun: Apply, args) =>
+            cpy.TypeApply(tree)(coverageTransform(fun), args)
+
+          // a.b
+          case Select(qual, name) =>
+            if qual.symbol.exists && qual.symbol.is(JavaDefined) then
+              // Java classes can't be used as a value, we can't instrument the
+              // qualifier ({<Probe>;System}.xyz() is not possible !), so instrument it
+              // as it is
+              instrument(tree)
+            else
+              val transformed = cpy.Select(tree)(coverageTransform(qual), name)
+              if transformed.qualifier.isDef then
+                // instrument calls to methods without parameter list
+                instrument(transformed)
+              else
+                transformed
+
+          case tree @ Match(sel, cases) =>
+            // force the cases if needed
+            cpy.Match(tree)(coverageTransform(sel), transformCases(cases, forceInstrument))
+
+          case tree: CaseDef =>
+            transformCaseDef(tree, forceInstrument)
+
+          case tree: ValDef =>
+            // only transform the rhs
+            val rhs = coverageTransform(tree.rhs)
+            cpy.ValDef(tree)(rhs=rhs)
+
+          case tree: DefDef =>
+            // only transform the params (for the default values) and the rhs
+            // force instrumentation of literals and other small trees in the rhs,
+            // to ensure that the method call are recorded
```

Comment:
What if instead we called `instrument(tree, tree.namePos)` there to record that the method was called?

Reply:
That would replace the DefDef by Block(call to Invoker.invoked, DefDef) :/

Reply:
Yes that's what I intended.